drivers/platform/xilinx/i2c.c

```diff
@@ -152,16 +214,55 @@ int32_t i2c_remove(struct i2c_desc *desc)
 int32_t i2c_write(struct i2c_desc *desc,
 		  uint8_t *data,
 		  uint8_t bytes_number,
-		  uint8_t stop_bit)
+		  uint8_t option)
 {
+	xil_i2c_desc	*xdesc;
+	int32_t		ret;
+
+	xdesc = desc->extra;
+
+	switch (xdesc->type) {
+	case IIC_PL:
 #ifdef XIIC_H
-	xil_i2c_desc *xil_desc;
-	xil_desc = desc->extra;
-	return XIic_Send(xil_desc->instance.BaseAddress, desc->slave_address, data,
-			 bytes_number, stop_bit ? XIIC_STOP : XIIC_REPEATED_START);
-#else
-	return SUCCESS;
+		ret = XIicPs_SetOptions(xdesc->instance,
```

Comment:
`XIicPs_SetOptions` is from the PS driver not PL. The PL equivalent would be `XIic_SetOptions` and it returns void, so the return value cannot be tested here. Check for the `i2c_write` function too.